lib/ansible/modules/cloud/amazon/iam_policy.py

```
@@ -112,240 +114,215 @@
-        elif state == 'absent':
+    def put(self, policy_doc):
+        if not self.check_mode:
+            self._put(self.name, self.policy_name, policy_doc)
+        self.changed = True
+
+    def _delete(self, name, policy_name):
+        pass
+
+    def delete(self):
+        self.changed = True
+        if not self.check_mode:
             try:
-                iam.delete_role_policy(name, policy_name)
-                changed = True
-            except boto.exception.BotoServerError as err:
-                error_msg = boto_exception(err)
-                if 'cannot be found.' in error_msg:
-                    changed = False
-                    module.exit_json(changed=changed,
-                                     msg="%s policy is already absent" % policy_name)
+                self._delete(self.name, self.policy_name)
+            except ClientError as e:
+                if e.response['Error']['Code'] == 'NoSuchEntity':
+                    self.changed = False
                 else:
-                    module.fail_json(msg=err.message)
+                    raise
 
-        updated_policies = [cp for cp in iam.list_role_policies(name).
-                            list_role_policies_result.
-                            policy_names]
-    except boto.exception.BotoServerError as err:
-        error_msg = boto_exception(err)
-        module.fail_json(changed=changed, msg=error_msg)
+    def get_policy_text(self):
+        if self.policy_document is not None:
+            return self.get_policy_from_document()
+        if self.policy_json is not None:
+            return self.get_policy_from_json()
+        return None
 
-    return changed, name, updated_policies
+    def get_policy_from_document(self):
+        try:
+            with open(self.policy_document, 'r') as json_data:
+                pdoc = json.dumps(json.load(json_data), sort_keys=True)
+                json_data.close()
+        except IOError as e:
+            if e.errno == 2:
```

If this if statement isn't true, it looks like the real error will be ignored.